app/models/user.rb

```diff
@@ -313,6 +313,45 @@ def compressed_export
     ActiveSupport::Gzip.compress Diaspora::Exporter.new(self).execute
   end
 
+  ######### Photos export ##################
+  mount_uploader :exported_photos_file, ExportedPhotos
+
+  def queue_export_photos
+    update exporting_photos: true
+    Workers::ExportPhotos.perform_async(id)
+  end
+
+  def perform_export_photos!
+    export = Tempfile.new([username, '_photos.zip'])
+    
+    Zip::ZipOutputStream.open(export.path) do |zos|
+      photos.each do |photo|
+        temp_photo = Tempfile.new(photo.remote_photo_name, encoding: 'ascii-8bit')
+        begin
+          photo_path = "#{Rails.root}/public/uploads/images/#{photo.remote_photo_name}"
```

Comment:
Sorry, but `Rails.root` does not give you the upload location or even the sites domain or anything, it gives you the absolute path on the filesystem where the application is located, so this still doesn't work for stuff stored on S3.

I would give you more pointers, but I'm myself very unsure about what the best way to handle that case would be.